clients/scheduler/src/components/time-slot-select/index.tsx

```diff
@@ -1,16 +1,68 @@
 import { h } from 'preact'
-import { ParsedSlotsType } from '@canvas-medical/embed-common'
-import { useState, useEffect } from 'preact/hooks'
+import { isSameDay, ParsedSlotsType } from '@canvas-medical/embed-common'
+import { useState, useEffect, useMemo, useCallback } from 'preact/hooks'
 import { useAppContext } from '../../hooks'
 import { Ui } from './ui'
 
 export const TimeSlotSelect = () => {
   const { fetchTimeSlots, date } = useAppContext()
-  const [timeSlots, setTimeSlots] = useState<ParsedSlotsType[]>([])
+  const [providerTimeSlots, setProviderTimeSlots] = useState<ParsedSlotsType[]>([])
+
+  const addTimeSlots = useCallback((newProviderTimeSlots: ParsedSlotsType[]) => {
+    const mergedProviderAvailability = newProviderTimeSlots.map((newProvider) => {
+      const provider = providerTimeSlots.find((entry) => entry.providerId === newProvider.providerId)
+      const mergedSlots = provider ? [...provider.providerSlots, ...newProvider.providerSlots] : newProvider.providerSlots
+      return {providerId: newProvider.providerId, providerSlots: mergedSlots}
+    })
+
+    setProviderTimeSlots(mergedProviderAvailability)
+  }, [providerTimeSlots])
```

Comment:
`providerTimeSlots` is provided as a dependency so that we can access the existing state, otherwise `providerTimeSlots` is always an empty array when we call this function which breaks backwards date search. 

Seems crazy to me that this doesn't cause an infinite loop 😅